classes/core/EntityDAO.inc.php

```diff
@@ -318,4 +319,20 @@ protected function convertToDB($value, string $type)
     {
         return $this->deprecatedDao->convertToDB($value, $type);
     }
+
+    /**
+     * Legacy function for grid handlers.
+     *Prepares LazyCollections in associative array GridHandler expects, e.g. [item_id => item]
+     *
+     * @see PKP\db\DAOResultFactory::toAssociativeArray()
+     *
+     */
+    public static function toAssociativeArray(LazyCollection $lazyCollection, string $idField = 'id'): array
+    {
+        $returner = [];
+        foreach ($lazyCollection as $item) {
+            $returner[$item->getData($idField)] = $item;
+        }
+        return $returner;
+    }
```

Comment:
@asmecher I think this was your code, right? I'd like to see this go into the base GridHandler class, rather than in EntityDAO. That way when we eventually remove grids we'll clean out the old code automatically. And the method won't need to be static.

Reply:
This harks to an old `DAOResultFactory` function, but is implemented for a `LazyCollection` instead. Yes, I agree, this should go into `GridHandler`.

Reply:
(On second thought: https://github.com/pkp/pkp-lib/pull/7195/files#r675648251)